src/ClientHandle.cpp

```diff
@@ -722,9 +722,10 @@ bool cClientHandle::HandleLogin(const AString & a_Username)
 		// Let the plugins know about this event, they may refuse the player:
 		if (cRoot::Get()->GetPluginManager()->CallHookLogin(*this, m_ProtocolVersion, a_Username))
 		{
-			Destroy();
+			SendDisconnect("Login Rejected!");
```

Comment:
I keep forgetting who handles what in clientHandle code. Does not calling destroy cause leaks?

Reply:
I believe that the Disconnect packet should be sent first. Once the player disconnects, the server automatically calls the `Destroy()` method. Other places where the player is kicked also use the `SendDisconnect` method.

Reply:
I see. This is not PR-specific but is the client's cooperation required? Can a malicious client ignore `SendDisconnect`, preventing player destruction?

Reply:
Yes, a malicious client could ignore the packet. I intend to fix the issue, as discussed below with @peterbell10.